### DISCUSSION TITLE
Can colima use a bridge network?

### DISCUSSION THREAD

**Discussion by bmwinstead** [abiosoft/colima#636] [Q&A]
Apologies if I get any terminology wrong...

With `colima start --network-address` I can trivially see the colima VM from my laptop, which is awesome! However, I'd _love_ it if I could have a bridge network or shared IP space that's shared between my laptop and the lima VM. 

The use case for this is to support installing MetalLB on a `kind` cluster and create IP-based load balancers that are visible from the host. I need to be able to have multiple LBs that use the same TCP port on different IPs, and this seems to be the only way to accomplish this. The only other way currently would be to use minikube with a vm driver and those are a little clunky... colima would make this a lot simpler.

Is this possible today? Using a lima override maybe? If not, is it a feasible feature request?

**Comment by leotu** [comment]
Ref. 
https://github.com/abiosoft/colima/issues/375#issuecomment-1659265171

Ref. Install socket_vmnet
https://lima-vm.io/docs/config/network/#socket_vmnet

**Comment by torarnv** [comment]
I have the same question. The instructions in https://github.com/abiosoft/colima/issues/375#issuecomment-1659265171 are not clear. 

The Lima override in `~/.lima/_config/override.yaml` is not picked up by Colima, as Colima seems to use a separate `$LIMA_HOME` in `~/.colima/_lima/`.

Modifying `~/.colima/_lima/_config/override.yaml` _does_ "work", in that the change is picked up, but since `~/.colima/_lima/_config/networks.yaml` only contains a `user-v2` network (https://github.com/abiosoft/colima/blob/9b0809d0ed9ad3ff1e57c405f27324e6298ca04f/embedded/network/networks.yaml) it fails with:

```> errors inspecting instance: [field `networks[2].lima` references network "bridged" which is not defined in networks.yaml]```

And the `networks.yaml` can't be edited, as it's overwritten each time Colima starts.

So how is this supposed to work?

**Comment by antonioberben** [comment]
@abiosoft any suggestion on this? ⬆️ 
How can we use bridged network? Thanks

**Comment by a0s** [comment]
Sooo, how to prevent overriding every time?
```shell
cat ~/.colima/_lima/_config/networks.yaml
# DO NOT EDIT!
# This file would be replaced by Colima on startup.

networks:
  user-v2:
    mode: user-v2
    gateway: 192.168.5.2
    netmask: 255.255.255.0
```

**Comment by abiosoft** [comment]
The main blocker for this is the ability to test effectively.

These network configurations sometimes mess up the machine and would require some cleanups. Especially during development when the process needs to be repeated severally. 
Unfortunately, due to lack of nested virtualization support for macOS guest OS, the only way to verify this is on my main development machine, which I am not willing to overly tamper with.

Until then, there is not much I can do, other than limiting the network options to what works for most users and easy for me to maintain.

**Comment by abiosoft** [comment]
The feature is being added https://github.com/abiosoft/colima/pull/1382.
